File: mobile/lib/features/operation_financiere/domain/enums/statut_operation.dart
enum StatutOperation { ENCAISSE, PAYE, ANNULEE }

extension StatutOperationExt on StatutOperation {
  String get libelle => switch (this) {
        StatutOperation.ENCAISSE => 'Encaissée',
        StatutOperation.PAYE     => 'Payée',
        StatutOperation.ANNULEE  => 'Annulée',
      };

  /// État terminal validé (encaissée ou payée).
  bool get estTerminee =>
      this == StatutOperation.ENCAISSE || this == StatutOperation.PAYE;

  static StatutOperation fromString(String v) {
    // Compat données legacy : les anciens statuts génériques deviennent ENCAISSE.
    if (v == 'VALIDEE' || v == 'BROUILLON') return StatutOperation.ENCAISSE;
    return StatutOperation.values.byName(v);
  }
}

File: backend/src/main/java/com/tmk/vtcmanager/application/domain/operation/StatutOperation.java
package com.tmk.vtcmanager.application.domain.operation;

public enum StatutOperation {
    ENCAISSE,
    PAYE,
    ANNULEE;

    /**
     * Statut "terminé" attendu selon le type d'opération :
     * un revenu est ENCAISSE, une dépense est PAYE.
     */
    public static StatutOperation termineePour(TypeOperation type) {
        return type == TypeOperation.REVENU ? ENCAISSE : PAYE;
    }

    /** Vrai si l'opération est dans un état terminal validé (encaissée/payée). */
    public boolean estTerminee() {
        return this == ENCAISSE || this == PAYE;
    }
}
